NYCPlanning/db-pluto
# create minor build pipeline

## changes
* add "minor release" option to github action which triggers data build
* create `01_data_loading_minor.sh` to use:
  * latest versions for minor release datasets
  * pinned versions for all other datasets (ideally, those used in last major release)
* add relevant versions to `version.env` and use them `01_data_loading_minor.sh`
* add "traps" to ensure shell scrips don't fail silently
* fix `DELETE TABLE` error in `config.sh/import_qaqc`
* fix missing `pluto_corrections` version in `source_data_versions.csv` output
* tweak issue template for data build (primarily used for tracking input data sources for major releases)
* update readme and fix broken badges

## tests
* [x] run minor release build GHA with Export/Upload as true ([completed](https://github.com/NYCPlanning/db-pluto/actions/runs/4057293969/jobs/6982860877))
* [x] confirm `pluto_corrections` version appears in `source_data_versions.csv`
* [x] make input versions the same as 22v3, run, and compare to ensure no differences ([completed](https://github.com/NYCPlanning/db-pluto/actions/runs/4059057243))
* [x] make major input versions the same as 22v3 (e.g. pluto_corrections) with minor input versions as latest (no declared version) and run ([completed](https://github.com/NYCPlanning/db-pluto/actions/runs/4064500564/jobs/6998011807))

## notes
* for minor builds, would be nice to get input dataset versions based solely on the major version by ingesting the relevant `source_data_versions.csv`. wouldn't work yet since PLUTO 22v3 has an empty record for which `pluto_corrections` version was used (it was 21v4)
